lib/msgfmt/msgfmt.go

```diff
@@ -140,6 +140,24 @@ func findUserInputEndIdx(userInputStartIdx int, msg []rune, userInput []rune) in
 	return msgIdx
 }
 
+// skipTrailingInputBoxLine checks if the next line contains all the given markers
+// and returns the incremented index if found. In case of Gemini and Cursor, the user
+// input is echoed back in a box. This function searches for the markers passed by the
+// caller and returns (currentIdx+1, true) if the next line contains all of them in the same order,
+// otherwise returns (currentIdx, false).
+func skipTrailingInputBoxLine(lines []string, currentIdx int, markers ...string) (idx int, found bool) {
+	if currentIdx+1 >= len(lines) {
+		return currentIdx, false
+	}
+	line := lines[currentIdx+1]
+	for _, m := range markers {
+		if !strings.Contains(line, m) {
+			return currentIdx, false
+		}
```

Comment:
I'm also relatively new to this codebase, so please let me know if I'm way off -- 
my understanding is that we want to match a "bottom line" of box-drawing characters.

This approach could potentially incorrectly match a line similar to:
```
Hello (╯▽╰) this is my custom agent 
```

Would a better approach be to check that the string trimmed of leading and trailing whitespace starts or ends with the given characters?

Reply:
Given that the pre-existing logic also uses `strings.Contains` it may make sense to consider this out of scope of the current PR. It's probably still worth investigating, so I'll create a follow-up issue.